### ISSUE TITLE
[TRACKER]Chef Workstation 2020 Testing - bootstrap testing on cross platform

### ISSUE THREAD

**Issue by dheerajd-msys** [chef/chef-workstation#1226]
## Description
- Setup chef-workstation on your system per https://github.com/chef/chef-workstation/issues/1167#issuecomment-627407517
- Verify functionality and leave a comment for the results of the command
## Parent task
https://github.com/chef/chef-workstation/issues/1167
## Reference
https://docs.chef.io/install_bootstrap/
https://docs.chef.io/workstation/knife_bootstrap/

**Comment by dheerajd-msys** [comment]
We've checked with on master for bootstrap.
1. **LINUX-->WINDOWS**
```
bundle exec knife
bootstrap 54.208.20.101 -N linux-windows --connection-user administrator --connection-
password "try824)9qQ?.5VbudpB&amp;dhtb=.8J)5!" -c ../stove/.chef/knife.rb -o winrm --run-list recipe[windows]

Connecting to 54.208.20.101

Creating new client for linux-windows
Creating new node for linux-windows
Bootstrapping 54.208.20.101
[54.208.20.101] Checking for existing directory &quot;C:\chef&quot;...
[54.208.20.101] Existing directory not found, creating.
[54.208.20.101]
[54.208.20.101] C:\Users\Administrator\Documents&gt;(
[54.208.20.101] echo.url = WScript.Arguments.Named(&quot;url&quot;)
[54.208.20.101] echo.path = WScript.Arguments.Named(&quot;path&quot;)
[54.208.20.101] echo.proxy = null
[54.208.20.101] echo.&#39;* Vaguely attempt to handle file:// scheme urls by url unescaping and
switching all
[54.208.20.101] echo.&#39;* / into . Also assume that file:/// is a local absolute path and that
file://&lt;foo&gt;
[54.208.20.101] echo.&#39;* is possibly a network file path.
[54.208.20.101] echo.If InStr(url, &quot;file://&quot;) = 1 Then
[54.208.20.101] echo.url = Unescape(url)
[54.208.20.101] echo.If InStr(url, &quot;file:///&quot;) = 1 Then
[54.208.20.101] echo.sourcePath = Mid(url, Len(&quot;file:///&quot;) + 1)
[54.208.20.101] echo.Else
[54.208.20.101] echo.sourcePath = Mid(url, Len(&quot;file:&quot;) + 1)
[54.208.20.101] echo.End If
[54.208.20.101] echo.sourcePath = Replace(sourcePath, &quot;/&quot;, &quot;\&quot;)
[54.208.20.101] echo.
[54.208.20.101] echo.Set objFSO = CreateObject(&quot;Scripting.FileSystemObject&quot;)
[54.208.20.101] echo.If objFSO.Fileexists(path) Then objFSO.DeleteFile path
[54.208.20.101] echo.objFSO.CopyFile sourcePath, path, true
[54.208.20.101] echo.Set objFSO = Nothing
[54.208.20.101] echo.
[54.208.20.101] echo.Else
[54.208.20.101] echo.Set objXMLHTTP = CreateObject(&quot;MSXML2.ServerXMLHTTP&quot;)
[54.208.20.101] echo.Set wshShell = CreateObject( &quot;WScript.Shell&quot; )
[54.208.20.101] echo.Set objUserVariables = wshShell.Environment(&quot;USER&quot;)
[54.208.20.101] echo.
[54.208.20.101] echo.rem http proxy is optional
[54.208.20.101] echo.rem attempt to read from HTTP_PROXY env var first
[54.208.20.101] echo.On Error Resume Next

[54.208.20.101] echo.
[54.208.20.101] echo.If NOT (objUserVariables(&quot;HTTP_PROXY&quot;) = &quot;&quot;) Then
[54.208.20.101] echo.proxy = objUserVariables(&quot;HTTP_PROXY&quot;)
[54.208.20.101] echo.
[54.208.20.101] echo.rem fall back to named arg
[54.208.20.101] echo.ElseIf NOT (WScript.Arguments.Named(&quot;proxy&quot;) = &quot;&quot;) Then
[54.208.20.101] echo.proxy = WScript.Arguments.Named(&quot;proxy&quot;)
[54.208.20.101] echo.End If
[54.208.20.101] echo.
[54.208.20.101] echo.If NOT isNull(proxy) Then
[54.208.20.101] echo.rem setProxy method is only available on ServerXMLHTTP 6.0+
[54.208.20.101] echo.Set objXMLHTTP = CreateObject(&quot;MSXML2.ServerXMLHTTP.6.0&quot;)
[54.208.20.101] echo.objXMLHTTP.setProxy 2, proxy
[54.208.20.101] echo.End If
[54.208.20.101] echo.
[54.208.20.101] echo.On Error Goto 0
[54.208.20.101] echo.
[54.208.20.101] echo.objXMLHTTP.open &quot;GET&quot;, url, false
[54.208.20.101] echo.objXMLHTTP.send()
[54.208.20.101] echo.If objXMLHTTP.Status = 200 Then
[54.208.20.101] echo.Set objADOStream = CreateObject(&quot;ADODB.Stream&quot;)
[54.208.20.101] echo.objADOStream.Open
[54.208.20.101] echo.objADOStream.Type = 1
[54.208.20.101] echo.objADOStream.Write objXMLHTTP.ResponseBody
[54.208.20.101] echo.objADOStream.Position = 0
[54.208.20.101] echo.Set objFSO = Createobject(&quot;Scripting.FileSystemObject&quot;)
[54.208.20.101] echo.If objFSO.Fileexists(path) Then objFSO.DeleteFile path
[54.208.20.101] echo.Set objFSO = Nothing
[54.208.20.101] echo.objADOStream.SaveToFile path
[54.208.20.101] echo.objADOStream.Close
[54.208.20.101] echo.Set objADOStream = Nothing
[54.208.20.101] echo.End If
[54.208.20.101] echo.Set objXMLHTTP = Nothing
[54.208.20.101] echo.End If
[54.208.20.101] ) 1&gt;C:\chef\wget.vbs
[54.208.20.101]
[54.208.20.101] C:\Users\Administrator\Documents&gt;(
[54.208.20.101] echo.param(
[54.208.20.101] echo. [String] $remoteUrl,
[54.208.20.101] echo. [String] $localPath
[54.208.20.101] echo.)
[54.208.20.101] echo.
[54.208.20.101] echo.$ProxyUrl = $env:http_proxy;
[54.208.20.101] echo.$webClient = new-object System.Net.WebClient;
[54.208.20.101] echo.
[54.208.20.101] echo.if ($ProxyUrl -ne &#39;&#39;) {
[54.208.20.101] echo. $WebProxy = New-Object System.Net.WebProxy($ProxyUrl,$true)
[54.208.20.101] echo. $WebClient.Proxy = $WebProxy
[54.208.20.101] echo.}

[54.208.20.101] echo.
[54.208.20.101] echo.$webClient.DownloadFile($remoteUrl, $localPath);
[54.208.20.101] ) 1&gt;C:\chef\wget.ps1
[54.208.20.101]
[54.208.20.101] C:\Users\Administrator\Documents&gt;(
[54.208.20.101]
[54.208.20.101]
[54.208.20.101]
[54.208.20.101] )
[54.208.20.101] Detected Windows Version 10.0 Build 17763
[54.208.20.101]
[54.208.20.101] C:\Users\Administrator\Documents&gt;goto Version10.0
[54.208.20.101]
[54.208.20.101] C:\Users\Administrator\Documents&gt;goto architecture_select
[54.208.20.101]
[54.208.20.101] C:\Users\Administrator\Documents&gt;IF &quot;AMD64&quot; == &quot;x86&quot; IF not defined
PROCESSOR_ARCHITEW6432
[54.208.20.101]
[54.208.20.101] C:\Users\Administrator\Documents&gt;goto chef_installed
[54.208.20.101] Checking for existing Chef Infra Client installation
[54.208.20.101]
[54.208.20.101] C:\Users\Administrator\Documents&gt;WHERE chef-client 1&gt;nul 2&gt;nul
[54.208.20.101]
[54.208.20.101] C:\Users\Administrator\Documents&gt;If !ERRORLEVEL! == 0 (
[54.208.20.101]
[54.208.20.101] goto key_create
[54.208.20.101] ) else (
[54.208.20.101]
[54.208.20.101] goto install
[54.208.20.101] )
[54.208.20.101] No existing installation of Chef Infra Client detected
[54.208.20.101] Checking for existing downloaded package at
&quot;C:\Users\ADMINI~1\AppData\Local\Temp\chef-client-latest.msi&quot;
[54.208.20.101] No existing downloaded packages to delete.
[54.208.20.101] Attempting to download client package using PowerShell if available...
[54.208.20.101] powershell.exe -ExecutionPolicy Unrestricted -InputFormat None -NoProfile -
NonInteractive -File C:\chef\wget.ps1
&quot;https://www.chef.io/chef/download?p=windows&amp;pv=2016&amp;m=x86_64&amp;DownloadContext=Po
werShell&amp;channel=stable&amp;v=16&quot; &quot;C:\Users\ADMINI~1\AppData\Local\Temp\chef-client-
latest.msi&quot;
[54.208.20.101] Download via PowerShell succeeded.
[54.208.20.101] Installing downloaded client package...
[54.208.20.101]
[54.208.20.101] C:\Users\Administrator\Documents&gt;msiexec /qn /log
&quot;C:\Users\ADMINI~1\AppData\Local\Temp\chef-client-msi29671.log&quot; /i
&quot;C:\Users\ADMINI~1\AppData\Local\Temp\chef-client-latest.msi&quot;
[54.208.20.101] Successfully installed Chef Infra Client package.
[54.208.20.101] Installation completed successfully
[54.208.20.101] Writing validation key...

[54.208.20.101] Validation key written.
[54.208.20.101]
[54.208.20.101] C:\Users\Administrator\Documents&gt;(
[54.208.20.101] echo.chef_server_url &quot;https://api.chef.io/organizations/dh&quot;
[54.208.20.101] echo.validation_client_name &quot;dh-validator&quot;
[54.208.20.101] echo.file_cache_path &quot;C:/chef/cache&quot;
[54.208.20.101] echo.file_backup_path &quot;C:/chef/backup&quot;
[54.208.20.101] echo.cache_options ({:path =&gt; &quot;C:\chef/cache/checksums&quot;, :skip_expires
=&gt; true})
[54.208.20.101] echo.chef_license &quot;accept&quot;
[54.208.20.101] echo.node_name &quot;linux-windows&quot;
[54.208.20.101] echo.log_level :auto
[54.208.20.101] echo.log_location STDOUT
[54.208.20.101] ) 1&gt;C:\chef\client.rb
[54.208.20.101]
[54.208.20.101] C:\Users\Administrator\Documents&gt;(echo.{&quot;run_list&quot;:[&quot;recipe[windows]&quot;]})
1&gt;C:\chef\first-boot.json
[54.208.20.101] Starting chef-client to bootstrap the node...
[54.208.20.101]
[54.208.20.101] C:\Users\Administrator\Documents&gt;SET
&quot;PATH=C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\Syste
m32\WindowsPowerShell\v1.0\;C:\ruby\bin;C:/opscode/chef\bin;C:/opscode/chef\embedded\b
in;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\
WindowsPowerShell\v1.0\;C:\Windows\System32\OpenSSH\;C:\Program Files\Amazon\cfn-
bootstrap\;C:\Windows\system32\config\systemprofile\AppData\Local\Microsoft\WindowsApps
;;;C:\Program Files\Docker;C:\Users\Administrator\AppData\Local\Microsoft\WindowsApps&quot;
[54.208.20.101]
[54.208.20.101] C:\Users\Administrator\Documents&gt;chef-client -c C:\chef/client.rb -j
C:\chef/first-boot.json
[54.208.20.101] +---------------------------------------------+
[54.208.20.101] âœ” 2 product licenses accepted.
[54.208.20.101] +---------------------------------------------+
[54.208.20.101] Starting Chef Infra Client, version 16.1.0
[54.208.20.101] resolving cookbooks for run list: [&quot;windows&quot;]
[54.208.20.101] Synchronizing Cookbooks:
[54.208.20.101] - windows (6.0.1)
[54.208.20.101] Installing Cookbook Gems:
[54.208.20.101] Compiling Cookbooks...
[54.208.20.101] [2020-05-25T11:14:56+00:00] WARN: Resource windows_user_privilege
from the client is overriding the resource from a cookbook. Please upgrade your cookbook or
remove the cookbook from your run_list.
[54.208.20.101] [2020-05-25T11:14:56+00:00] WARN: The windows::default recipe has been
deprecated. The gems previously installed in this recipe ship in the Chef MSI.
[54.208.20.101] Converging 0 resources
[54.208.20.101]
[54.208.20.101] Running handlers:
[54.208.20.101] Running handlers complete
[54.208.20.101] Chef Infra Client finished, 0/0 resources updated in 18 seconds
```

**Comment by dheerajd-msys** [comment]
2. **LINUX-->LINUX**
=================
```
bundle exec knife
bootstrap 3.83.227.143 -U ubuntu --sudo --ssh-identity-file ../habitat/brocode.pem -N
linuxtolinuxaws --run-list recipe[myhaproxy]  -c ../stove/.chef/knife.rb
Connecting to 3.83.227.143
The authenticity of host &#39;3.83.227.143 ()&#39; can&#39;t be established.
fingerprint is SHA256:nHmPrx2NNVfRcI2YSAGu3UUmt3/WX1kWHP2styTvQ3w.
Are you sure you want to continue connecting
? (Y/N) Y
Connecting to 3.83.227.143
Creating new client for linuxtolinuxaws
Creating new node for linuxtolinuxaws
Bootstrapping 3.83.227.143
[3.83.227.143] -----&gt; Installing Chef Omnibus (stable/16)
downloading https://omnitruck.chef.io/chef/install.sh
to file /tmp/install.sh.1871/install.sh
[3.83.227.143] trying wget...
[3.83.227.143] ubuntu 18.04 x86_64
Getting information for chef stable 16 for ubuntu...
downloading
https://omnitruck.chef.io/stable/chef/metadata?v=16&amp;p=ubuntu&amp;pv=18.04&amp;m=x86_64
to file /tmp/install.sh.1875/metadata.txt
[3.83.227.143] trying wget...
[3.83.227.143] sha1 53f187ff85678c9504731827147d638a0ab120b1
sha256 9995a251dc82298a7badc19e881cd3b1c55f7be8fdbe09691170c000b61ec0e2
url https://packages.chef.io/files/stable/chef/16.1.0/ubuntu/18.04/chef_16.1.0-1_amd64.deb
version 16.1.0
[3.83.227.143]
[3.83.227.143] downloaded metadata file looks valid...
[3.83.227.143] downloading
https://packages.chef.io/files/stable/chef/16.1.0/ubuntu/18.04/chef_16.1.0-1_amd64.deb
to file /tmp/install.sh.1875/chef_16.1.0-1_amd64.deb
[3.83.227.143] trying wget...
[3.83.227.143] Comparing checksum with sha256sum...
[3.83.227.143] Installing chef 16
installing with dpkg...
[3.83.227.143] Selecting previously unselected package chef.
[3.83.227.143] (Reading database ... 56588 files and directories currently installed.)

[3.83.227.143] Preparing to unpack .../chef_16.1.0-1_amd64.deb ...
[3.83.227.143] Unpacking chef (16.1.0-1) ...
[3.83.227.143] Setting up chef (16.1.0-1) ...
[3.83.227.143] Thank you for installing Chef Infra Client! For help getting started visit
https://learn.chef.io
[3.83.227.143] Starting the first Chef Infra Client Client run...
[3.83.227.143] +---------------------------------------------+
✔ 2 product licenses accepted.
+---------------------------------------------+
[3.83.227.143] Starting Chef Infra Client, version 16.1.0
[3.83.227.143]
[3.83.227.143] resolving cookbooks for run list: [&quot;myhaproxy&quot;]
[3.83.227.143]
[3.83.227.143] Synchronizing Cookbooks:
[3.83.227.143]
[3.83.227.143] - myhaproxy (0.15.0)
[3.83.227.143] - haproxy (8.3.0)
[3.83.227.143] - build-essential (8.2.1)
[3.83.227.143] - seven_zip (3.1.2)
[3.83.227.143] - windows (6.0.1)
[3.83.227.143]
[3.83.227.143] - mingw (2.1.0)
[3.83.227.143]
[3.83.227.143]
[3.83.227.143] - yum-epel (3.3.0)
[3.83.227.143]
[3.83.227.143] Installing Cookbook Gems:
[3.83.227.143] Compiling Cookbooks...
[3.83.227.143]
[3.83.227.143] [2020-05-25T12:29:46+00:00] WARN: Resource windows_user_privilege from
the client is overriding the resource from a cookbook. Please upgrade your cookbook or
remove the cookbook from your run_list.
[3.83.227.143] Converging 7 resources
[3.83.227.143]
[3.83.227.143] Recipe: myhaproxy::server
[3.83.227.143]
[3.83.227.143]
[3.83.227.143] * apt_update[] action periodic
[3.83.227.143]
[3.83.227.143]
[3.83.227.143] * directory[/var/lib/apt/periodic] action create
[3.83.227.143] (up to date)
[3.83.227.143]
[3.83.227.143]
[3.83.227.143] * directory[/etc/apt/apt.conf.d] action create
[3.83.227.143] (up to date)
[3.83.227.143]
[3.83.227.143]
[3.83.227.143] * file[/etc/apt/apt.conf.d/15update-stamp] action create_if_missing

[3.83.227.143] (up to date)
[3.83.227.143]
[3.83.227.143]
[3.83.227.143] * execute[apt-get -q update] action run
[3.83.227.143]
- execute [&quot;apt-get&quot;, &quot;-q&quot;, &quot;update&quot;]
- update new lists of packages
* haproxy_install[package] action create
[3.83.227.143]
[3.83.227.143]
[3.83.227.143] * apt_package[haproxy] action install
[3.83.227.143]
- install version 1.8.8-1ubuntu0.10 of package haproxy
[3.83.227.143]
* group[haproxy] action create (up to date)
* linux_user[haproxy] action create
[3.83.227.143]
- alter user haproxy
* directory[/etc/haproxy] action create
[3.83.227.143]
[3.83.227.143]
[3.83.227.143] - change owner from &#39;root&#39; to &#39;haproxy&#39;
[3.83.227.143]
[3.83.227.143]
[3.83.227.143] - change group from &#39;root&#39; to &#39;haproxy&#39;
[3.83.227.143]
[3.83.227.143]
[3.83.227.143] * template[/etc/haproxy/haproxy.cfg] action nothing
[3.83.227.143] (skipped due to action :nothing)
[3.83.227.143]
[3.83.227.143]
[3.83.227.143] * haproxy_config_global[] action create
[3.83.227.143] (up to date)
[3.83.227.143]
[3.83.227.143]
[3.83.227.143] * haproxy_config_defaults[defaults] action create
[3.83.227.143] (up to date)
[3.83.227.143]
[3.83.227.143]
[3.83.227.143] * haproxy_frontend[http-in] action create
[3.83.227.143] (up to date)
[3.83.227.143]
[3.83.227.143]
[3.83.227.143] * haproxy_backend[servers] action create
[3.83.227.143] (up to date)
[3.83.227.143]
[3.83.227.143]
[3.83.227.143] * haproxy_service[haproxy] action create
[3.83.227.143] (up to date)

[3.83.227.143]
[3.83.227.143]
[3.83.227.143] * cookbook_file[/etc/default/haproxy] action create
[3.83.227.143]
[3.83.227.143]
[3.83.227.143] - update content in file /etc/default/haproxy from c316bc to e3b0c4
[3.83.227.143]
[3.83.227.143]
[3.83.227.143] --- /etc/default/haproxy 2019-10-28 10:59:19.000000000 +0000
[3.83.227.143]
[3.83.227.143]
[3.83.227.143] +++ /etc/default/.chef-haproxy20200525-1952-1bh0zh 2020-05-25
12:30:00.344175957 +0000
[3.83.227.143]
[3.83.227.143]
[3.83.227.143] @@ -1,11 +1 @@
[3.83.227.143]
[3.83.227.143]
[3.83.227.143] -# Defaults file for HAProxy
[3.83.227.143]
[3.83.227.143]
[3.83.227.143] -#
[3.83.227.143]
[3.83.227.143]
[3.83.227.143] -# This is sourced by both, the initscript and the systemd unit file, so do not
[3.83.227.143]
[3.83.227.143]
[3.83.227.143] -# treat it as a shell script fragment.
[3.83.227.143]
[3.83.227.143]
[3.83.227.143] -
[3.83.227.143]
[3.83.227.143]
[3.83.227.143] -# Change the config file location if needed
[3.83.227.143]
[3.83.227.143]
[3.83.227.143] -#CONFIG=&quot;/etc/haproxy/haproxy.cfg&quot;
[3.83.227.143]
[3.83.227.143]
[3.83.227.143] -
[3.83.227.143]
[3.83.227.143]
[3.83.227.143] -# Add extra flags here, see haproxy(1) for a few options
[3.83.227.143]
[3.83.227.143]
[3.83.227.143] -#EXTRAOPTS=&quot;-de -m 16&quot;
[3.83.227.143]
[3.83.227.143]
[3.83.227.143] * systemd_unit[haproxy.service] action create

[3.83.227.143]
[3.83.227.143]
[3.83.227.143] * file[/etc/systemd/system/haproxy.service] action create
[3.83.227.143]
- create new file /etc/systemd/system/haproxy.service
- update content in file /etc/systemd/system/haproxy.service from none to d181d6
--- /etc/systemd/system/haproxy.service 2020-05-25 12:30:00.568184502 +0000
+++ /etc/systemd/system/.chef-haproxy20200525-1952-em4fpq.service 2020-05-25
12:30:00.500181908 +0000
@@ -1 +1,19 @@
+[Unit]
+Description=HAProxy Load Balancer
+Documentation=file:/usr/share/doc/haproxy/configuration.txt.gz
+After=network.target syslog.service
+[Service]
+EnvironmentFile=-/etc/default/haproxy
+Environment=&quot;CONFIG=/etc/haproxy/haproxy.cfg&quot; &quot;PIDFILE=/run/haproxy.pid&quot;
+ExecStartPre=/usr/sbin/haproxy -f $CONFIG -c -q
+ExecStart=/usr/sbin/haproxy -Ws -f $CONFIG -p $PIDFILE $OPTIONS
+ExecReload=/usr/sbin/haproxy -f $CONFIG -c -q
+ExecReload=/bin/kill -USR2 $MAINPID
+KillSignal=TERM
+User=root
+WorkingDirectory=/
+KillMode=mixed
+Restart=always
+[Install]
+WantedBy=multi-user.target
[3.83.227.143]
[3.83.227.143]
[3.83.227.143] - change mode from &#39;&#39; to &#39;0644&#39;
[3.83.227.143]
[3.83.227.143]
[3.83.227.143] - change owner from &#39;&#39; to &#39;root&#39;
[3.83.227.143]
[3.83.227.143]
[3.83.227.143] - change group from &#39;&#39; to &#39;root&#39;
[3.83.227.143]
[3.83.227.143] - creating unit: haproxy.service
* systemd_unit[haproxy.service] action enable
[3.83.227.143] (up to date)
* service[haproxy] action nothing (skipped due to action :nothing)
* template[/etc/haproxy/haproxy.cfg] action nothing (skipped due to action :nothing)
* template[/etc/haproxy/haproxy.cfg] action create
[3.83.227.143]
[3.83.227.143]
[3.83.227.143] - update content in file /etc/haproxy/haproxy.cfg from 2076c5 to 3bfb7b
[3.83.227.143]
[3.83.227.143]

[3.83.227.143] - suppressed sensitive resource
[3.83.227.143]
[3.83.227.143]
[3.83.227.143] - change owner from &#39;root&#39; to &#39;haproxy&#39;
[3.83.227.143]
[3.83.227.143]
[3.83.227.143] - change group from &#39;root&#39; to &#39;haproxy&#39;
[3.83.227.143]
[3.83.227.143]
[3.83.227.143] * service[haproxy] action enable
[3.83.227.143] (up to date)
* service[haproxy] action start
[3.83.227.143] (up to date)
* service[haproxy] action restart
[3.83.227.143]
- restart service service[haproxy]
[3.83.227.143]
[3.83.227.143] Running handlers:
[3.83.227.143]
[3.83.227.143] Running handlers complete
[3.83.227.143] Chef Infra Client finished, 11/26 resources updated in 17 seconds
[3.83.227.143]
```

**Comment by Vasu1105** [comment]
Bootstrapping from Windows workstation fails as mentioned in https://github.com/chef/chef/issues/9914

**Comment by dheerajd-msys** [comment]
Bootstrapping from windows to windows is also getting failed as mentioned in https://github.com/chef/chef/issues/9844#issuecomment-633348195. We've also reproduced this from our end.

**Comment by dheerajd-msys** [comment]
https://github.com/chef/chef/issues/9914 is fixed in https://github.com/chef/chef/pull/9984

**Comment by dheerajd-msys** [comment]
chef/chef#9922 is fixed in chef/chef#10298

**Comment by dheerajd-msys** [comment]
**Windows--->Windows**
==========================
```
PS C:\Users\Administrator\chef> bundle exec knife bootstrap 18.216.167.39 -U Administrator -P "ewMd;v%ZvQiB%e)?ezFmgndM%
;H5*IqI" -N ddubey-win19-w2w-bootstrap -o winrm -c "C:\Users\Administrator\Downloads\chef-starter\chef-repo\.chef\knife.
rb"


Connecting to 18.216.167.39 using winrm
WARNING: Performing legacy client registration with the validation key at C:/Users/Administrator/Downloads/chef-starter/
chef-repo/.chef/msystech-validator.pem...
WARNING: Remove the key file or remove the 'validation_key' configuration option from your config.rb (knife.rb) to use m
ore secure user credentials for client registration.
Bootstrapping 18.216.167.39
 [18.216.167.39] Checking for existing directory "C:\chef"...
 [18.216.167.39] Existing directory found, skipping creation.
 [18.216.167.39]
 [18.216.167.39] C:\Users\Administrator\Documents>(
 [18.216.167.39] echo.url = WScript.Arguments.Named("url")
 [18.216.167.39]  echo.path = WScript.Arguments.Named("path")
 [18.216.167.39]  echo.proxy = null
 [18.216.167.39]  echo.'* Vaguely attempt to handle file:// scheme urls by url unescaping and switching all
 [18.216.167.39]  echo.'* / into .  Also assume that file:/// is a local absolute path and that file://<foo>
 [18.216.167.39]  echo.'* is possibly a network file path.
 [18.216.167.39]  echo.If InStr(url, "file://") = 1 Then
 [18.216.167.39]  echo.url = Unescape(url)
 [18.216.167.39]  echo.If InStr(url, "file:///") = 1 Then
 [18.216.167.39]  echo.sourcePath = Mid(url, Len("file:///") + 1)
 [18.216.167.39]  echo.Else
 [18.216.167.39]  echo.sourcePath = Mid(url, Len("file:") + 1)
 [18.216.167.39]  echo.End If
 [18.216.167.39]  echo.sourcePath = Replace(sourcePath, "/", "\")
 [18.216.167.39]  echo.
 [18.216.167.39]  echo.Set objFSO = CreateObject("Scripting.FileSystemObject")
 [18.216.167.39]  echo.If objFSO.Fileexists(path) Then objFSO.DeleteFile path
 [18.216.167.39]  echo.objFSO.CopyFile sourcePath, path, true
 [18.216.167.39]  echo.Set objFSO = Nothing
 [18.216.167.39]  echo.
 [18.216.167.39]  echo.Else
 [18.216.167.39]  echo.Set objXMLHTTP = CreateObject("MSXML2.ServerXMLHTTP")
 [18.216.167.39]  echo.Set wshShell = CreateObject( "WScript.Shell" )
 [18.216.167.39]  echo.Set objUserVariables = wshShell.Environment("USER")
 [18.216.167.39]  echo.
 [18.216.167.39]  echo.rem http proxy is optional
 [18.216.167.39]  echo.rem attempt to read from HTTP_PROXY env var first
 [18.216.167.39]  echo.On Error Resume Next
 [18.216.167.39]  echo.
 [18.216.167.39]  echo.If NOT (objUserVariables("HTTP_PROXY") = "") Then
 [18.216.167.39]  echo.proxy = objUserVariables("HTTP_PROXY")
 [18.216.167.39]  echo.
 [18.216.167.39]  echo.rem fall back to named arg
 [18.216.167.39]  echo.ElseIf NOT (WScript.Arguments.Named("proxy") = "") Then
 [18.216.167.39]  echo.proxy = WScript.Arguments.Named("proxy")
 [18.216.167.39]  echo.End If
 [18.216.167.39]  echo.
 [18.216.167.39]  echo.If NOT isNull(proxy) Then
 [18.216.167.39]  echo.rem setProxy method is only available on ServerXMLHTTP 6.0+
 [18.216.167.39]  echo.Set objXMLHTTP = CreateObject("MSXML2.ServerXMLHTTP.6.0")
 [18.216.167.39]  echo.objXMLHTTP.setProxy 2, proxy
 [18.216.167.39]  echo.End If
 [18.216.167.39]  echo.
 [18.216.167.39]  echo.On Error Goto 0
 [18.216.167.39]  echo.
 [18.216.167.39]  echo.objXMLHTTP.open "GET", url, false
 [18.216.167.39]  echo.objXMLHTTP.send()
 [18.216.167.39]  echo.If objXMLHTTP.Status = 200 Then
 [18.216.167.39]  echo.Set objADOStream = CreateObject("ADODB.Stream")
 [18.216.167.39]  echo.objADOStream.Open
 [18.216.167.39]  echo.objADOStream.Type = 1
 [18.216.167.39]  echo.objADOStream.Write objXMLHTTP.ResponseBody
 [18.216.167.39]  echo.objADOStream.Position = 0
 [18.216.167.39]  echo.Set objFSO = Createobject("Scripting.FileSystemObject")
 [18.216.167.39]  echo.If objFSO.Fileexists(path) Then objFSO.DeleteFile path
 [18.216.167.39]  echo.Set objFSO = Nothing
 [18.216.167.39]  echo.objADOStream.SaveToFile path
 [18.216.167.39]  echo.objADOStream.Close
 [18.216.167.39]  echo.Set objADOStream = Nothing
 [18.216.167.39]  echo.End If
 [18.216.167.39]  echo.Set objXMLHTTP = Nothing
 [18.216.167.39]  echo.End If
 [18.216.167.39] ) 1>C:\chef\wget.vbs
 [18.216.167.39]
 [18.216.167.39] C:\Users\Administrator\Documents>(
 [18.216.167.39] echo.param(
 [18.216.167.39]  echo.   [String] $remoteUrl,
 [18.216.167.39]  echo.   [String] $localPath
 [18.216.167.39]  echo.)
 [18.216.167.39]  echo.
 [18.216.167.39]  echo.$ProxyUrl = $env:http_proxy;
 [18.216.167.39]  echo.$webClient = new-object System.Net.WebClient;
 [18.216.167.39]  echo.
 [18.216.167.39]  echo.if ($ProxyUrl -ne '') {
 [18.216.167.39]  echo.  $WebProxy = New-Object System.Net.WebProxy($ProxyUrl,$true)
 [18.216.167.39]  echo.  $WebClient.Proxy = $WebProxy
 [18.216.167.39]  echo.}
 [18.216.167.39]  echo.
 [18.216.167.39]  echo.$webClient.DownloadFile($remoteUrl, $localPath);
 [18.216.167.39] ) 1>C:\chef\wget.ps1
 [18.216.167.39]
 [18.216.167.39] C:\Users\Administrator\Documents>(
 [18.216.167.39]
 [18.216.167.39]
 [18.216.167.39]
 [18.216.167.39] )
 [18.216.167.39] Detected Windows Version 10.0 Build 17763
 [18.216.167.39]
 [18.216.167.39] C:\Users\Administrator\Documents>goto Version10.0
 [18.216.167.39]
 [18.216.167.39] C:\Users\Administrator\Documents>goto architecture_select
 [18.216.167.39]
 [18.216.167.39] C:\Users\Administrator\Documents>IF "AMD64" == "x86" IF not defined PROCESSOR_ARCHITEW6432
 [18.216.167.39]
 [18.216.167.39] C:\Users\Administrator\Documents>goto chef_installed
 [18.216.167.39] Checking for existing Chef Infra Client installation
 [18.216.167.39]
 [18.216.167.39] C:\Users\Administrator\Documents>WHERE chef-client  1>nul 2>nul
 [18.216.167.39]
 [18.216.167.39] C:\Users\Administrator\Documents>If !ERRORLEVEL! == 0 (
 [18.216.167.39]
 [18.216.167.39]  goto key_create
 [18.216.167.39] )  else (
 [18.216.167.39]
 [18.216.167.39]  goto install
 [18.216.167.39] )
 [18.216.167.39] Existing Chef Infra Client installation detected, skipping download
 [18.216.167.39] Writing validation key...
 [18.216.167.39] Validation key written.
 [18.216.167.39]
 [18.216.167.39] C:\Users\Administrator\Documents>(
 [18.216.167.39] echo.chef_server_url  "https://api.chef.io/organizations/msystech"
 [18.216.167.39]  echo.validation_client_name "msystech-validator"
 [18.216.167.39]  echo.file_cache_path   "C:\\chef\\cache"
 [18.216.167.39]  echo.file_backup_path  "C:\\chef\\backup"
 [18.216.167.39]  echo.cache_options     ({:path => "C:\\chef\\cache\\checksums", :skip_expires => true})
 [18.216.167.39]  echo.chef_license "accept"
 [18.216.167.39]  echo.node_name "ddubey-win19-w2w-bootstrap"
 [18.216.167.39]  echo.log_level        :auto
 [18.216.167.39]  echo.log_location       STDOUT
 [18.216.167.39] ) 1>C:\chef\client.rb
 [18.216.167.39]
 [18.216.167.39] C:\Users\Administrator\Documents>(echo.{"run_list":[]}) 1>C:\chef\first-boot.json
 [18.216.167.39] Starting chef-client to bootstrap the node...
 [18.216.167.39]
 [18.216.167.39] C:\Users\Administrator\Documents>SET "PATH=;C:\Windows;\Wbem;\WindowsPowerShell\v1.0\;C:\ruby\bin;C:\op
scode\chef\bin;C:\opscode\chef\embedded\bin;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\
WindowsPowerShell\v1.0\;C:\Windows\System32\OpenSSH\;C:\Program Files\Amazon\cfn-bootstrap\;C:\opscode\chef\bin\;C:\User
s\Administrator\AppData\Local\Microsoft\WindowsApps"
 [18.216.167.39]
 [18.216.167.39] C:\Users\Administrator\Documents>chef-client -c C:\chef\client.rb -j C:\chef\first-boot.json
 [18.216.167.39] Starting Chef Infra Client, version 16.4.35
 [18.216.167.39] [2020-08-18T09:37:08+00:00] WARN: Plugin Network: unable to detect ipaddress
 [18.216.167.39] resolving cookbooks for run list: []
 [18.216.167.39] Synchronizing Cookbooks:
 [18.216.167.39] Installing Cookbook Gems:
 [18.216.167.39] Compiling Cookbooks...
 [18.216.167.39] [2020-08-18T09:37:14+00:00] WARN: Node ddubey-win19-w2w-bootstrap has an empty run list.
 [18.216.167.39] Converging 0 resources
 [18.216.167.39]
 [18.216.167.39] Running handlers:
 [18.216.167.39] Running handlers complete
 [18.216.167.39] Chef Infra Client finished, 0/0 resources updated in 07 seconds
PS C:\Users\Administrator\chef>
```

**Comment by mcascone** [comment]
Amazing work everyone! How will we know what version of the latest&greatest workstation we can pick up to get this?
